### DISCUSSION TITLE
[Chatper 11] 자동 장애 복구? 수동 장애 복구?

### DISCUSSION THREAD

**Discussion by Irisation23** [cowaine/spring-boot-for-msa#184] [Chapter 11. Spring Scheduling Tasks]
> 11.3 배치 서버 아키텍처 하단 참조

```
장애에 견고한 코드를 작성하여 실행되지 못한 태스크는 자동으로 복구할
수 있도록 작성한다. 짧은 주기의 배치 프로세스가 아니라면 수동으로 복구할 수 있는 기능을 개발한다.
```

배치프로세스의 데이터를 처리함에 있어 안정적이라면 큰 무리 없겠지만, 만약 반드시 기록되어야 할 정보가 기록되지 못한채 유실된다면 이는 기업의 손해로 예상이 되는데요 그렇다면 안정적인 장애 복구를 위한 방법은 어떤것이 있을지 고민해 봤습니다.

실행되지 못한 짧은 주기의 배치 프로세스를 수동으로 복구할 방법은 log로 기록된 정보를 참조해 이를 일괄적으로 수동으로 처리할 수 있는 방법이 떠오릅니다.

그러나 
### Q. 자동으로 복구할 방법은 Retry 시도 이외에 또 다른 자동 복구할 수 있는 방법이 있을까요?

영업 비밀일까요? 궁금합니다.

**Comment by JoisFe** [comment]
ㅋㅋㅋ 영업 비밀

**Comment by corock** [comment]
_cf._ [[NHN FORWARD 22] 편안한 휴식 시간을 지켜줄 안정적인 백엔드 운영과 개발 기법](https://www.youtube.com/watch?v=2hCbY_mpqSg&list=PL42XJKPNDepYXyKefvicxlA2fz1aThVs5&index=3)
